frontends/web/src/components/bitboxbase/unlock.tsx

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright 2019 Shift Devices AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, h, RenderableProps } from 'preact';
+import { alertUser } from '../../components/alert/Alert';
+import { Button } from '../../components/forms';
+import { Entry } from '../../components/guide/entry';
+import { Guide } from '../../components/guide/guide';
+import { AppLogo, SwissMadeOpenSource } from '../../components/icon/logo';
+import { Footer, Header } from '../../components/layout';
+import { PasswordSingleInput } from '../../components/password';
+import { translate, TranslateProps } from '../../decorators/translate';
+import { apiPost } from '../../utils/request';
+
+interface UnlockBaseProps {
+    bitboxBaseID: string | null;
+}
+
+interface State {
+    bitboxBaseID: string | null;
```

Comment:
we settled on `undefined` vs `null` some time back, so `bitboxBaseID?: string` is better.